### PR TITLE
Commerce docs improvements

### PR DESCRIPTION
**Pull request by lukeholder**

### CODE REVIEW

File: customers.md [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
+Commerce will consolidate customers by email address. A person may checkout as a guest customer with the same email multiple times, and a new guest customer will be created each time. After an order is completed, customers are consolidated so there is only ever one customer per email address.
 
-After a user is registered and verified, whenever they log in, all guest orders that used the email address belonging to that user are transferred to the logged in user. This means, even if a registered user makes an anonymous/guest order, the next time they log in, the order will appear in their order history, as long as they used the same email address.
+Once a user is registered and verified, orders on their email address will be consolidated whenever they log in. Any guest orders with the user’s email address are transferred to the user. As a result, even future anonymous/guest orders with that email address will appear in the user’s order history.
```
Comment (lukeholder) [inline customers.md]: This has changed in Commerce 3 and we haven‘t updated this.

How it works is the following:

All registered users in the system have an associated customer record. Then when a logged in user is building a cart on the front end, the customerId of the order/cart is always the customerId belonging to that user.

If a guest customer creates an order, a new guest customer is created for that order. When they complete that order, if there was a registered user with the same email address as the order, the order is moved to belong to the customerId of the registered user.
If there was not a registered user with the same email, but their was a previous guest customer with the same email address, the order is moved to belong to that single guest customer.

Everything now happens on order completion.

The above system means that guest customers that had their orders move to another customer, are now orphaned, and we garbage collect them (delete them) often.

let me know if you need more clarification @mattstein

Comment (mattstein) [reply customers.md]: Thanks Luke! This makes sense to me except for one part:

> All registered users in the system have an associated customer record.

Are you saying that a) every User now inherently has an associated Customer by default, or that b) every visitor that’s created a store account has a Customer record with an associated Craft user (as it was before)?

It seems like you’ve mostly described a change to the Customer consolidation strategy that’s now orphan-and-garbage-collect, and that the moment of tidying up is now on order completion and not login. Is that accurate?

Comment (lukeholder) [reply customers.md]: In Commerce 3, anytime a Craft user element is saved, we make sure a related customer record exists for that user, and create one for that user if there wasn't one.

In the Commerce 3 migration (developers moving from commerce 2) we generated a customer record for every Craft user element that didnt have one.


> It seems like you’ve mostly described a change to the Customer consolidation strategy that’s now orphan-and-garbage-collect, and that the moment of tidying up is now on order completion and not login. Is that accurate?

Exactly.

Comment (mattstein) [reply customers.md]: Cool, then it sounds like it’s _a)_, where a tight User<->Customer relationship is a careful design choice in v3. And the Craft User has a Customer record even if they never created an order.

This is all making sense, thanks for clarifying!

**Comment by mattstein** [comment]
@lukeholder I can’t designate you as a reviewer since I guess this is your PR, but I’ve finished going through all the pages and I’m ready for whatever feedback you’ve got!

File: adding-to-and-updating-the-cart.md [hunk 2]
```diff
@@ -118,28 +119,28 @@ You can add multiple purchasables to the cart in an update cart form. You supply
...
+        <input type="hidden" name="purchasables[{{ loop.index }}][qty]" value="1">
+        <input type="hidden" name="purchasables[{{ loop.index }}][note]" value="1">
     {% endfor %}
 
     <input type="submit" value="Add all variants to cart">
 </form>
 ```
 
-While using multi-add the same rules apply for updating a quantity vs adding to cart, based on the uniquessness of the options `signature` and `purchasableId`.
+While using multi-add the same rules apply for updating a quantity vs. adding to cart, based on the uniquessness of the options `signature` and `purchasableId`.
```
Comment (lukeholder) [inline adding-to-and-updating-the-cart.md:131]: The attribute on the line item is called `optionsSignature` which then hash of the `options` on the line item.

File: adding-to-and-updating-the-cart.md [hunk 3]
```diff
@@ -118,28 +119,28 @@ You can add multiple purchasables to the cart in an update cart form. You supply
...
 Once the purchasable has been added to the cart, your customer may want to update the `qty` or `note`, they can do this by updating a line item.
 
-Line items can have their `qty`, `note`, and `options`updated. They can also be removed.
+Line items can have their `qty`, `note`, and `options` updated or removed.
 
-To update a line item, submit a form array param with the name of `lineItems`, with the ID of the array key being the line item ID.
+To update a line item, submit a `lineItems` form array, where each item’s array key is the line item ID.
 
-Example:
+Example using `LINE_ITEM_ID` as a placeholder for the ID of the item you’d like to edit:
```
Comment (lukeholder) [inline adding-to-and-updating-the-cart.md:143]: Lets get rid of the `LINE_ITEM_ID` in all the examples, and just add a loop over light items within the example and use `{{ item.id }}`

File: customer-address-management.md [hunk 4]
```diff
@@ -4,54 +4,117 @@ When a customer checks out with a new address, the address is added to their add
...
+{% if address %}
+  {# gets all validation errors for the address #}
+  {% set errors = address.getErrors() %}
+
+  {# gets the `firstName` error for the address #}
+  {% set firstNameErrors = address.getErrors('firstName') %}
+{% endif %}
+```
+
+For a complete template example that outputs individual field validation errors, see [shop/\_includes/addresses/form.html](https://github.com/craftcms/commerce/blob/develop/templates/shop/_includes/addresses/form.html) in the [example templates](example-templates.md).
```
Comment (lukeholder) [inline customer-address-management.md:120]: is the `\` a typo in `shop/\_includes/addresses/form.html`?

File: twig-filters.md [hunk 5]
```diff
@@ -1,44 +1,59 @@
...
+| Parameter    | Type   | Default | Required | Description                                                                                                                                      |
+| ------------ | ------ | ------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `currency`   | string |         | yes      | Valid, three-character ISO payment currency set up in Commerce → Store Settings → Payment Currencies.                                            |
+| `convert`    | bool   | `false` | no       | Should the amount passed to this filter be converted to `currency`’s exchange rate?                                                              |
+| `format`     | bool   | `true`  | no       | Should the amount passed to this filter be formatted according to `currency`? (Typically adds currency symbol and thousands+decimal separators.) |
+| `stripZeros` | bool   | `false` | no       | Should trailing zeros (`.00`) be stripped from a formatted number?                                                                               |
```
Comment (lukeholder) [inline twig-filters.md:12]: Tables are sooo much better! Good job.

**Pull request merged**